src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.FloatingValueSet.cs

```diff
@@ -49,6 +50,23 @@ internal static IValueSet<TFloating> Random(int expectedSize, Random random)
 
             bool IValueSet.IsEmpty => !_hasNaN && _numbers.IsEmpty;
 
+            ConstantValue IValueSet.Sample
+            {
+                get
+                {
+                    if (!_numbers.IsEmpty)
+                    {
+                        var sample = _numbers.Sample;
+                        Debug.Assert(sample is { });
+                        return sample;
+                    }
+
+                    Debug.Assert(_hasNaN);
```

Comment:
Do we expect this to throw in Release configuration if `!_hasNaN && _numbers.IsEmpty`?

Reply:
The doc for `IValueSet.Sample` says it should throw `ArgumentException`. Will add code to do that.

---
In reply to: [441169310](https://github.com/dotnet/roslyn/pull/44702#discussion_r441169310) [](ancestors = 441169310)